docs/README.md

```diff
@@ -676,15 +676,112 @@ In this chapter you'll walk through these concepts and its details.
+├── src
+│   ├── commands
+│   ├── common
+│   ├── config
+│   ├── entities
+│   ├── events <------ They must be here
+│   ├── index.ts
+│   └── read-models
+```
+
 #### Creating events
 
-#### The event store
+The preferred way to create event files is the `new:event` generator, e.g.
+
+```shell
+boost new:event StockMoved --fields productID:string origin:string destination:string quantity:number
+```
+
+That will generate a file called `StockMoved.ts` under the proper `project-root/src/events` directory. You can also create the file manually, but we recommend using the generator and avoid dealing manually with boilerplate code.
+
+Note:
+
+> Running the event generator with an EventName already existing, will override the content of the current one. Soon, we will display a warning before overwriting anything. Meantime, if you missed a field, just add it to the class because, in Booster, all the infrastructure and data structures are inferred from your code.
+
+#### Registering events and the event store
+
+Creating an event file is different than storing an event instance in the event store. Booster calls The last `registering` an event. As said before, Booster applications are event-sourced, which means that all the events are stored forever. Think that all events go to a kind of event store, the store is a log queried by the [reducer functions](#4-entities-and-reducers) to recreate the application's current state.
+
+Booster injects the register as a parameter in the `handle` method of both the command and the event handlers. Then, registering an event is as simple as calling this function `register.events([event1, event2, ...])`, e.g.
+
+##### Registering events from command handlers
+
+```typescript
+@Command({
+  authorize: [Admin],
+})
+export class MoveStock {
+  public constructor(
+    readonly productID: string,
+    readonly origin: string,
+    readonly destination: string,
+    readonly quantity: number
+  ) {}
+
+  public async handle(register: Register): Promise<void> {
+    if (!this.enoughStock(this.origin, this.quantity, this.productID)) {
+      register.events(new ErrorEvent(`There is not enough stock for ${this.productID} at ${this.origin}`))
+    }
+  }
+}
+```
+
+##### Registering events from event handlers
+
+In the case of the event handlers, you also receive the event instance that triggered the handle function.
+
+```typescript
+@EventHandler(StockMoved)
+export class HandleAvailability {
+  public static async handle(event: StockMoved, register: Register): Promise<void> {
+    if (event.origin == 'provider') {
+      register.events(new ProductAvailabilityChanged(event.productID, event.quantity))
+    } else if (event.destination == 'customer') {
+      register.events(new ProductAvailabilityChanged(event.productID, -event.quantity))
+    }
+  }
+}
+```
 
 #### Events ordering
 
+<!-- TODO: several people have asked about how Booster ensures event ordering. I think it makes sense to explain that here  -->
```

Comment:
Will this bit be added as a future task or in this PR?

Reply:
In a different PR, I'll create a ticket because I don't know how we ensure events ordering

Reply:
It's ensured by sharding events by entity name and ID, but definitely, create a ticket so any of us can elaborate!